projection/core/src/main/scala/com/lightbend/lagom/projection/State.scala

```diff
@@ -65,6 +112,7 @@ final class State(val projections: Seq[Projection]) extends ProjectionSerializab
   def findWorker(workerKey: String): Option[Worker] =
     projections.flatMap(_.workers).find(_.key == workerKey)
 
+  /** Java API  */
```

Comment:
I just notice that maybe we need `findWorker`/`findProjection` APIs for Java (in terms of `Optional`, I mean. Maybe:

```scala
def getProjection(projectionName: String): Optional[Projection]
def getWorker(workerKey: String): Optional[Worker]
```

WDYT?

Reply:
Yeah, I noticed that we were missing those methods too, and TBH, I'm not sure whether we should add the java ones or remove (or make private) the scala ones.

I'll give a second thought.

Reply:
Yeah, the more I think about it the less I like those two methods being public.

1. `def getProjection` is trivial to implement by users, and
2. I don't like `def getWorker` getting a `workerKey:String` as an argument. Instead I'd like something more typed like `WorkerCoordinates` but that's not a public class so maybe `def getWorker(projectionName:String, tagName:String)`. In any case, the implementation is trivial too.

Reply:
I just removed the methods and moved to a couple of places (e.g.tests) where they were being used for convenience.

This keeps the API surface smaller.